lib/Dialect/FIRRTL/Transforms/IMDeadCodeElim.cpp

```diff
@@ -399,6 +406,46 @@ void IMDeadCodeElimPass::rewriteModuleSignature(FModuleOp module) {
   numRemovedPorts += deadPortIndexes.size();
 }
 
+void IMDeadCodeElimPass::eraseEmptyModule(FModuleOp module) {
+  // Public modules cannot be erased.
+  if (module.isPublic())
+    return;
+
+  // If the module doesn't have arguments, operations or annotations, we
+  // consider it to be dead.
+  if (!module.getBody()->args_empty() || !module.getBody()->empty() ||
+      !module.annotations().empty())
+    return;
+
+  // If there is "output_file" attribute, we don't delete the module.
```

Comment:
Not sure about this.  Seems acceptable.

Reply:
That makes sense. I was afraid that some metadata refers output file names before IMDCE pass run. I'll delete this condition after making sure that it doesn't break builds.